packages/integration-tests/projects/suite-web/tests/settings/custom-firmware.test.ts

```diff
@@ -0,0 +1,49 @@
+// @group:settings
+// @retry=2
+describe('Install custom firmware', () => {
+    beforeEach(() => {
+        cy.task('startEmu', { wipe: true });
+        cy.task('setupEmu');
+        cy.task('startBridge');
+        cy.viewport(1080, 1440).resetDb();
+        cy.prefixedVisit('/settings/device');
+        cy.passThroughInitialRun();
+    });
+    /*
+    *1. Navigate to `Settings/Device`
+    *2. Click on `Install firmware`
+    *3. Select the custom firmware
+        a)Observe the initialization on the device
+    *4. Complete the FW instalation on the device
+    */
+
+    it('go to device settings and check if custom FW modal appears', () => {
+        //
+        // Test preparation
+        //
+        cy.getTestElement('@settings/device/custom-firmware-modal-button').click({
+            force: true,
+        });
+        //
+        // Test execution
+        //
+        cy.get('[data-test="@firmware-custom"]').then(fileUploadModal => {
+            cy.wrap(fileUploadModal)
+                .find('input[type=file]')
+                .attachFile('test.bin', { subjectType: 'drag-n-drop' });
+        });
+        cy.get('[data-test="@firmware-custom"]').then(installFWbutton => {
+            cy.wrap(installFWbutton)
+                .find('[class*="Button"]')
+                .should('not.be.disabled')
+                .contains('Install firmware')
+                .click();
+        });
```

Comment:
idea for improvement: would be great to add some screenshots here. You can use 
`cy.screenshot(name)` which does not do any assertions, only produces screenshots or `cy.matchImageSnapshots(name)`

Reply:
@STew790 , after a discussion with @mroz22, we agreed that for sanity tests, it does not make sense to make and maintain screenshots as there is usually only one assert. So when dealing with sanity tests, let's keep using chai assertions unless there is a very good individual use case.